manifests/osfamily/darwin.pp

```diff
@@ -1,19 +1,25 @@
 class puppet_agent::osfamily::darwin{
   assert_private()
 
+  if $::macosx_productversion_major =~ /^11\./ {
```

Comment:
Assuming Apple will bump the major version on every macOS release, can we reverse the logic?
```suggestion
  if $::macosx_productversion_major =~ /^10\./ {
```

Also, `macosx_productversion_major` will be just `11` because of https://github.com/puppetlabs/facter/pull/2365. Since facter 3 doesn't have this change, maybe we can check based on `macosx_productversion` which should have the full version string everytime. Anyway if we reverse the logic it should be fine.